shared/plugins/setupBuild.gradle

```diff
@@ -21,9 +21,19 @@ if (!project.hasProperty('onlylinuxathena')) {
                     }
                 }
                 binaries.all {
+                    // Define a custom entry point depending on whether are building as a static library
+                    // or a shared module.
                     if (it instanceof StaticLibraryBinarySpec) {
-                        it.buildable = false
-                        return
+                        // These scenario is seldom used, except in sysid to build a fully static exe
+                        // with simulation modules. When building static, it is important that no two
+                        // modules have the same entry point symbolically.
+                        def name = project.name.replace("halsim_", "").toUpperCase()
+                        it.cppCompiler.define("HALSIM_ENTRY_POINT", "HALSIM_InitExtension_$name")
+                    } else {
+                        // This is the most common scenario. Modules can have the same entry point as
+                        // it makes it easier for the dlopen() logic. Furthermore, each module is loaded
+                        // in its own address space, so symbols from 2 dlopen()ed modules cannot clash.
+                        it.cppCompiler.define("HALSIM_ENTRY_POINT", "HALSIM_InitExtension")
```

Comment:
You could avoid defining anything at all in the dynamic case by leaving the cpp files as InitExtension and then redefining InitExtension only in the static case.